### ISSUE TITLE
sha3 test vectors don't match up with other implementations

### ISSUE THREAD

**Issue by doomrobo** [philanc/plc#3]
Running Python's `hashlib.sha3` on the same inputs gives outputs that disagree with the plc implementation's output. Python's output also happens to agree with this other online implementation: https://emn178.github.io/online-tools/sha3_512.html

```python
from hashlib import sha3_512, sha3_256

print(sha3_512(b"").digest().hex())
print(sha3_512(b"abc").digest().hex())
print(sha3_512(b"1" * 128).digest().hex())

print(sha3_256(b"").digest().hex())
print(sha3_256(b"abc").digest().hex())
print(sha3_256(b"1" * 128).digest().hex())

# Output
# a69f73cca23a9ac5c8b567dc185a756e97c982164fe25859e0d1dcc1475c80a615b2123af1f5f94c11e3e9402c3ac558f500199d95b6d3e301758586281dcd26
# b751850b1a57168a5693cd924b6b096e08f621827444f70d884f5d0240d2712e10e116e9192af3c91a7ec57647e3934057340b4cf408d5a56592f8274eec53f0
# 1e996f94a2c54cacd0fb3a778a3605e34c928bb9f51ffed970f05919dabf2fa3fe12d5eafcb8457169846c67b5e30ede9b22c4c59ace3c11663965e4dba28294
# a7ffc6f8bf1ed76651c14756a061d662f580ff4de43b49fa82d80a4b80f8434a
# 3a985da74fe225b2045c172d6bd390bd855f086e3e9d525b46bfe24511431532
# fdfe32511b76f38a71a7ac95a4c98add2b41debab35e1b7dda8bb3b14c280533
```

Notably, Python's output for the sha2 test vectors actually do match up with plc's, so there doesn't seem to be a systemic incompatibility here.

**Comment by philanc** [comment]
Thanks for the report. I will look into as soon as I can.

**Comment by philanc** [comment]
Hi Michael,  Just found the bug. 

I actually fell in the trap described here [1]. The sha3.lua code (and the web calculator I tested against!) used to implement the original Keccak submission to SHA-3. The final SHA-3 standard has been slightly modified: The first padding byte (0x01) in Keccak has been replaced with 0x06 in the final SHA-3.

The code in sha3.lua and test vectors in test_sha3.lua have been fixed. Many thanks for your report.
